src/Callback.php

```diff
@@ -194,19 +194,16 @@ public function handle_order_ready_for_capture_callback( $ledyer_order_id, $ledy
 		);
 
 		Ledyer_Payments()->logger()->debug( '[CALLBACK]: process order ready for capture callback: ' . $ledyer_order_id, $context );
-
-		$orders = wc_get_orders(
-			array(
-				'meta_query'   => array(
-					array(
-						'key'     => '_wc_ledyer_order_id',
-						'value'   => $ledyer_order_id,
-						'compare' => '=',
-					),
-				),
-				'date_created' => '>' . ( time() - MONTH_IN_SECONDS ),
-			)
+		$args   = array(
+			'meta_key'     => '_wc_ledyer_order_id', // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+			'meta_value'   => $ledyer_order_id, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+			'meta_compare' => '=',
+			'order'        => 'DESC',
+			'orderby'      => 'date',
+			'limit'        => 1,
+			'date_created' => '>' . ( time() - MONTH_IN_SECONDS ),
 		);
+		$orders = wc_get_orders( $args );
 
 		$order_id = isset( $orders[0] ) ? $orders[0]->get_id() : null;
```

Comment:
Before using the order here, we should add a check to ensure the order is actually the correct one. We have had issues in the past where the query fails, but it results in returning the last order placed instead. So we should verify this by checking that the order we get, actually has the same meta value stored. You can see [here](https://github.com/krokedil/klarna-checkout-for-woocommerce/blob/master/includes/kco-functions.php#L948-L954) how we do that in KCO. We also do this in another place in this plugin [here](https://github.com/krokedil/ledyer-payments-for-woocommerce/blob/80586482f4673c9d089d40ee26eab5567ea39da7/src/Gateway.php#L439-L441).